theLudovyc/TheFactory.gd

```diff
@@ -2,159 +2,80 @@ extends Node
 class_name TheFactory
 
 @onready var game: Game2D = get_parent()
-@onready var storage = $"../TheStorage"
-@onready var event_bus = $"../EventBus"
+@onready var storage: TheStorage = $"../TheStorage"
+@onready var event_bus: EventBus = $"../EventBus"
 
-enum Production_Line { current_workers, production_rate, current_ticks }
-var production_lines = {}
+var buildings: Array[Building2D] = []
 
 var workers := 0:
 	set(value):
 		workers = value
 		event_bus.available_workers_updated.emit(game.population - workers)
 
-enum Waiting_Lines { resource_type, needed_workers }
-var waiting_lines = []
+var waiting_line_buildings: Array[Building2D] = []
```

Comment:
I like the way you use building here. But please respect the MVC architecture

Reply:
Done